API/Converters/AssetConverter.cs

```diff
@@ -0,0 +1,286 @@
+                asset.Duration = hydraImage.Duration.Value;
+            }
+
+            if (hydraImage.Error != null)
+            {
+                asset.Error = hydraImage.Error;
+            }
+
+            if (hydraImage.Family != null)
+            {
+                asset.Family = (DLCS.Model.Assets.AssetFamily)hydraImage.Family;
+            }
+
+            if (hydraImage.Finished != null)
+            {
+                asset.Finished = hydraImage.Finished;
+            }
+
+            if (hydraImage.Width != null)
+            {
+                asset.Width = hydraImage.Width.Value;
+            }
+            if (hydraImage.Height != null)
+            {
+                asset.Height = hydraImage.Height.Value;
+            }
+
+            if (hydraImage.Ingesting != null)
+            {
+                asset.Ingesting = hydraImage.Ingesting.Value;
+            }
+
+            if (hydraImage.Origin != null)
+            {
+                asset.Origin = hydraImage.Origin;
+            }
+
+            if (hydraImage.String1 != null)
+            {
+                asset.Reference1 = hydraImage.String1;
+            }
+            if (hydraImage.String2 != null)
+            {
+                asset.Reference2 = hydraImage.String2;
+            }
+            if (hydraImage.String3 != null)
+            {
+                asset.Reference3 = hydraImage.String3;
+            }
+
+            if (hydraImage.Number1 != null)
+            {
+                asset.NumberReference1 = hydraImage.Number1.Value;
+            }
+            if (hydraImage.Number2 != null)
+            {
+                asset.NumberReference2 = hydraImage.Number2.Value;
+            }
+            if (hydraImage.Number3 != null)
+            {
+                asset.NumberReference3 = hydraImage.Number3.Value;
+            }
+
+            if (hydraImage.Roles != null)
+            {
+                asset.RolesList = hydraImage.Roles;
+            }
+            if (hydraImage.Tags != null)
+            {
+                asset.TagsList = hydraImage.Tags;
+            }
+
+            if (hydraImage.MaxUnauthorised != null)
+            {
+                asset.MaxUnauthorised = hydraImage.MaxUnauthorised.Value;
+            }
+
+            if (hydraImage.MediaType != null)
+            {
+                asset.MediaType = hydraImage.MediaType;
+            }
+
+            var thumbnailPolicy = hydraImage.ThumbnailPolicy.GetLastPathElement("thumbnailPolicies/");
+            if (thumbnailPolicy != null)
+            {
+                asset.ThumbnailPolicy = thumbnailPolicy;
+            }
+            
+            var imageOptimisationPolicy = hydraImage.ImageOptimisationPolicy.GetLastPathElement("imageOptimisationPolicies/");
+            if (imageOptimisationPolicy != null)
+            {
+                asset.ImageOptimisationPolicy = imageOptimisationPolicy;
+            }
+            
+            // Not patchable? Does Engine set this?
```

Comment:
IIRC in Deliverator `InitialOrigin` can only be set on the initial creation - it's passed to Engine, encoded in the SQS message. Engine will use it for ingestion if present. That's all it's used for, it's not persisted anywhere.

Reply:
Now also sent to engine (with test)